platform/linux-generic/odp_ipsec.c

```diff
@@ -1396,6 +1396,9 @@ static int ipsec_random_data(uint8_t *data, uint32_t len)
 	return 0;
 }
 
+/*
+ * Generate cipher IV for outbound processing.
+ */
 static int ipsec_out_iv(ipsec_state_t *state,
```

Comment:
Yes, I thought about it but didn't do it in this patch set. I am not sure if it would be good to put the function pointer directly in the SA instead of an SA type specific structure that could then hold other type specific data and functions too.

Reply:
Type specific structure could work too. Would need some prototyping and comparison.